src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt

```diff
@@ -331,7 +336,12 @@ class AutoImportFix(element: RsElement) : LocalQuickFixOnPsiElement(element), Hi
             ) ?: return false
             val element = path.reference.deepResolve() as? RsQualifiedNamedElement ?: return false
             if (!context.namespaceFilter(element)) return false
-            return !(element.parent is RsMembers && element.ancestorStrict<RsTraitItem>() != null)
+
+            // Looks like it's useless to access trait associated types directly (i.e. `Trait::Type`),
+            // but methods can be used in UFCS and associated functions or constants can be accessed
+            // it they have `Self` type in a signature
+            return element !is RsAbstractable || element.owner !is RsAbstractableOwner.Trait ||
+                element.canBeAccessedByTraitName || path.qualifier?.reference?.deepResolve() !is RsTraitItem
```

Comment:
What do you think about splitting out this condition into several ones (for example, `element !is RsAbstractable || element.owner !is RsAbstractableOwner.Trait`, `element.canBeAccessedByTraitName` and `path.qualifier?.reference?.deepResolve() !is RsTraitItem`)?
It should simplify debugging and separate light logic checks (`element !is RsAbstractable || element.owner !is RsAbstractableOwner.Trait`) from heavy ones: type inference (`element.canBeAccessedByTraitName`) and name resolution (`path.qualifier?.reference?.deepResolve() !is RsTraitItem`). Of course, the last reason is just a matter of taste